src/metal/metal-device.cpp

```
@@ -490,6 +492,80 @@ std::span<MTL::Resource* const> DeviceImpl::getAccelerationStructureResources()
+
+    TextureDesc desc = fixupTextureDesc(desc_);
+    NS::SharedPtr<MTL::TextureDescriptor> textureDesc = NS::TransferPtr(MTL::TextureDescriptor::alloc()->init());
+    switch (desc.memoryType)
+    {
+    case MemoryType::DeviceLocal:
+        textureDesc->setStorageMode(MTL::StorageModePrivate);
+        break;
+    case MemoryType::Upload:
+        textureDesc->setStorageMode(MTL::StorageModeShared);
+        textureDesc->setCpuCacheMode(MTL::CPUCacheModeWriteCombined);
+        break;
```

Cppcheck (2.21.0)

[style] The function 'shouldDeferPipelineCompilation' is never used.

(unusedFunction)